metabase/metabase
# Too Many Errors

## Background

In the course of development, I crash the application a lot. We have a very helpful top-level error boundary component that catches errors, and tries to reload the application. Unfortunately, if the app is in an unrecoverable state, this causes an infinite reload of the errored components that makes the app completely unresponsive and eventually just crashes the browser.

This is simply annoying in development, but what is more concerning is that something like an errored dashboard in production infinitely reloading could absolutely hammer a client's data warehouse and drive up their bills.

## Description

This sets a limit of 3 caught errors by the errorBoundary component before the application simply stops reloading a shows an error page.

![Screen Shot 2022-09-21 at 11 00 36 AM](https://user-images.githubusercontent.com/30528226/191567007-7b1eb77e-6316-492a-8ca6-459c65c53131.png)


## Testing Steps

Put a `throw new Error("test");` in the render method of something like `DashCard.jsx` and load up a dashboard. See that it errors only a finite number of times, rather than forever.
